autoload/db/adapter/presto.vim

```diff
@@ -0,0 +1,79 @@
+if exists('g:autoloaded_db_presto')
+  finish
+endif
+let g:autoloaded_db_presto = 1
+
+function! s:command_for_url(params) abort
+  let cmd = 'presto'
+  for [k, v] in items(a:params)
+    let cmd .= ' --'.k.' '.v
+  endfor
+  return cmd
+endfunction
+
+function! s:params(url) abort
+  let params = db#url#parse(a:url)
+  let presto_params = { 'server': 'localhost' }
+  if has_key(params, 'host')
+    let presto_params.server = params.host
+  endif
+  if has_key(params, 'port')
+    let presto_params.server .= ':'.params.port 
+  endif
+  if has_key(params, 'path')
+    let path = split(params.path, '/')
+    if len(path) >= 1
+      let presto_params.catalog = path[0]
+    end
+    if len(path) == 2
+      let presto_params.schema = path[1]
+    endif
+  endif
+  return presto_params
+endfunction
+
+function! db#adapter#presto#interactive(url) abort
+  return s:command_for_url(s:params(a:url))
+endfunction
+
+function! db#adapter#presto#input_flag() abort
+  return '--output-format ALIGNED --file '
+endfunction
+
+function! db#adapter#presto#complete_opaque(url) abort
+  let prefix = ''
+  let params = s:params(a:url)
+  let num_params = len(params)
+  let base = '//'.params.server
+  if (num_params == 1 && a:url =~ '/$') || (num_params == 2 && a:url !~ '/$')
```

Comment:
Is it possible to replace these with `has_key()` checks? Worried about what happens when additional parameters get added.